src/meta/src/manager/catalog/mod.rs

```diff
@@ -156,13 +157,51 @@ pub struct CatalogManager {
 pub struct CatalogManagerCore {
     pub database: DatabaseManager,
     pub user: UserManager,
+    /// `DdlController` will update this map, and pass the `tx` side to `CatalogController`.
+    /// On notifying, we can remove the entry from this map.
+    pub table_id_to_tx: HashMap<TableId, Sender<MetaResult<NotificationVersion>>>,
+    /// Catalogs which were marked as finished and committed.
+    /// But the `DdlController` has not instantiated notification channel.
+    /// Once notified, we can remove the entry from this map.
+    pub table_id_to_version: HashMap<TableId, MetaResult<NotificationVersion>>,
```

Comment:
I think we don't need this `table_id_to_version` to store the pending info. `table_id_to_tx` should be enough. We can check whether a streaming job has finished by directly checking job status in the `DatabaseManager`.

Ideally things can work like this
1. when register a tx on a table id, acquire lock and check the job status. If job is creating, store the tx. If the job has been created, return with ok. If the job does not exist, return err to tell the user about it. 
2. On `abort_all`, notify all tx with err.
3. Every drop/cancel streaming job ddl should remove the tx of the dropped/cancelled streaming job id.

Reply:
We also need to propagate the result and notify the version still, because FE will wait on this version. Knowing that a job is done creating is insufficient.

It exists because the catalog manager may commit before stream manager calls `wait_streaming_job`. In such a scenario, catalog manager needs to still notify the stream manager of the version and error.

Reply:
> We also need to propagate the result and notify the version still, because FE will wait on this version. Knowing that a job is done creating is insufficient.

Shall we just return the global latest notification version instead of the version at the moment the job turned created, so that we don't need to store the version per table. cc @yezizp2012 

> It exists because the catalog manager may commit before stream manager calls wait_streaming_job. In such a scenario, catalog manager needs to still notify the stream manager of the version and error.

I think this corner case can be covered in my comment above. If the catalog manager has commit before `wait_streaming_job`, there are 3 possibilities:
* job has been created, and then we just return ok, with the global latest notification version
* failed to create the first barrier. Stream manager should have got an error before calling `wait_streaming_job`, no need to handle.
* First barrier collected, and job status must have been turned into `Creating`. Even though there can be error and recovery happened, we can just store the tx anyway. It will be notified either `Ok` when the job finishes backfill afterward, or `Err` when a later error triggers a recovery and call `abort_all`.